### DISCUSSION TITLE
Alert reporting

### DISCUSSION THREAD

**Discussion by stijnpieters** [Security-Onion-Solutions/securityonion#4151] [Unsupported Versions]
I'm looking to do some weekly reporting of the alerts received throughout the week.

I know reporting is on the roadmap, as seen in https://github.com/Security-Onion-Solutions/securityonion/discussions/2534

However, I found a "way" to do some kind of basic reporting. I created an ElastAlert rule in which ElastAlert creates an aggregated table of all alerts seen in the configured schedule (e.g. in the last week).

My only problem now is formatting. The aggregation summary is formatted using a monospace font, however I can't configure this in an ElastAlert rule, this has to be done client-side. I tried passing a <style> block in the alert's body, but that required html formatting in the alert, and that messed up the layout of the aggregation summary.

I found an alternate way, that should be able to "beautify" the table: https://github.com/datacratic/elastalert/blob/datacratic/elastalert/emailjinja.py
But it requires a python library that is not in the ElastAlert container (util).

How would I go about adding this library to the container, if at all possible?

alert for reference:
[weekly_report.yaml.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6451212/weekly_report.yaml.txt)

EDIT:
it would seem util is in the install directory of elastalert. My elastalert container however still crashes when trying to import the custom "emailjinja.py"

**Comment by weslambert** [comment]
The most efficient way to add the library to the container would be to use your own custom image -- otherwise, you would need to install the library every time the container would be built.  Alternatively, you could bind mount the static files to the container so they are always available, or you could use Kibana's reporting engine instead.  I noticed we discussed this a while back in another post -- is there a reason you do not want to pursue that type of approach (Elastic reporting)?
